apps/dashboard/app/helpers/projects_helper.rb

```diff
@@ -31,4 +31,28 @@ def button_category(status)
       status
     end
   end
+
+  # TODO: Use it to populate drop down selection of shared directory in import project form
+  def cached_shared_dir
+    Rails.cache.fetch('cached_shared_dir', expires_in: 1.hour) do
+      shared_dir_list
+    end
+  end
+  
+  private
+  def shared_dir_list
+    shared_path = []
+    Configuration.shared_projects_root.each do |dir_path|
+      next unless File.readable?(dir_path)
+      Dir.each_child(dir_path) do |child|
+        child_dir = File.join(dir_path, child)
+        gid = File.stat(child_dir).gid
+        name = Etc.getgrgid(gid).name rescue nil
+        next unless CurrentUser.group_names.include?(name)
+        shared_path << child_dir
+      end
+    end
+
+    return shared_path
+  end
```

Comment:
1. I think these should resolve the Project since we have from_directory. That is, the return value should be an array of Project objects.
2. We don't need to rely on the group name matching the directory name. If the directory is readable, attempt to create a Project, otherwise move on.
3. See my note below about Ruby and using temporary variables like `shared_path`.

Reply:
okay I agree. Let me try to fix these.